src/Mod/Path/PathScripts/PathDrilling.py

```diff
@@ -23,130 +23,235 @@
-        elif obj.DwellEnabled and obj.DwellTime > 0:
-            cmd = "G82"
-            cmdParams['P'] = obj.DwellTime
-
-        # parentJob = PathUtils.findParentJob(obj)
-        # startHeight = obj.StartDepth.Value + parentJob.SetupSheet.SafeHeightOffset.Value
         startHeight = obj.StartDepth.Value + self.job.SetupSheet.SafeHeightOffset.Value
 
-        for p in holes:
-            params = {}
-            params['X'] = p['x']
-            params['Y'] = p['y']
+        edgelist = []
+        for hole in holes:
+            v1 = FreeCAD.Vector(hole["x"], hole["y"], obj.StartDepth.Value)
+            v2 = FreeCAD.Vector(hole["x"], hole["y"], obj.FinalDepth.Value - endoffset)
+            edgelist.append(Part.makeLine(v1, v2))
+
+        # iterate the edgelist and generate gcode
+        for edge in edgelist:
+
+            PathLog.debug(edge)
 
             # move to hole location
-            self.commandlist.append(Path.Command('G0', {'X': p['x'], 'Y': p['y'], 'F': self.horizRapid}))
-            self.commandlist.append(Path.Command('G0', {'Z': startHeight, 'F': self.vertRapid}))
-            self.commandlist.append(Path.Command('G1', {'Z': obj.StartDepth.Value, 'F': self.vertFeed}))
 
-            # Update changes to parameters
-            params.update(cmdParams)
+            command = Path.Command(
+                "G0", {"X": hole["x"], "Y": hole["y"], "F": self.horizRapid}
+            )
+            self.commandlist.append(command)
+            machine.addCommand(command)
 
-            # Perform canned drilling cycle
-            self.commandlist.append(Path.Command(cmd, params))
+            command = Path.Command("G0", {"Z": startHeight, "F": self.vertRapid})
+            self.commandlist.append(command)
+            machine.addCommand(command)
 
-            # Cancel canned drilling cycle
-            self.commandlist.append(Path.Command('G80'))
-            self.commandlist.append(Path.Command('G0', {'Z': obj.SafeHeight.Value}))
+            command = Path.Command(
+                "G1", {"Z": obj.StartDepth.Value, "F": self.vertFeed}
+            )
+            self.commandlist.append(command)
+            machine.addCommand(command)
+
+            # Technical Debt:  We are assuming the edges are aligned.
+            # This assumption should be corrected and the necessary rotations
+            # performed to align the edge with the Z axis for drilling
+
+            # Perform drilling
+            dwelltime = obj.DwellTime if obj.DwellEnabled else 0.0
+            peckdepth = obj.PeckDepth.Value if obj.PeckEnabled else 0.0
+            repeat = 1  # technical debt:  Add a repeat property for user control
+
+            try:
+                drillcommands = generator.generate(edge, dwelltime, peckdepth, repeat)
+
+            except ValueError as e:  # any targets that fail the generator are ignored
+                PathLog.info(e)
+                continue
+
+            self.commandlist.extend(drillcommands)
```

Comment:
Should there be a loop here to add each command also to the machine state?

Reply:
Yes, that's a good point.  That's going to take some thought.  The XY words can definitely affect position.  The Z word is the drilling depth but the position where the machine ends up after the move is the old Z prior to the move, not the Z coordinate.

Reply:
Updated PathMachineState to handle drillcommands correctly. 
implemented the loop here as suggested